### PR TITLE
remote: add note about sys.exit behaviour

### PR DESCRIPTION
**Pull request by oliver-sanders**
Add a note about `sys.exit` behaviour in the remote interfaces.

### CODE REVIEW

File: cylc/flow/remote.py [hunk 1]
```diff
@@ -107,6 +107,10 @@ def run_cmd(
         * Else True if the remote command is executed successfully, or
           if unsuccessful and capture_status=True the remote command exit code.
         * Otherwise exit with an error message.
+
+    Exits:
+        1: In the event of certain command errors.
```
Comment (MetRonnie) [inline cylc/flow/remote.py:112]: I think `Exits with code 1 in the event of certain command errors.` might be a little clearer

**Review by MetRonnie — APPROVED** [review]

**Review by hjoliver — APPROVED** [review]

File: cylc/flow/remote.py (hunk 1 above)
Comment (hjoliver) [reply cylc/flow/remote.py:112]: And/or "Exit codes" rather than "Exits".

Comment (oliver-sanders) [reply cylc/flow/remote.py:112]: All yours!

Comment (MetRonnie) [reply cylc/flow/remote.py:112]:
```suggestion
    Exits with code 1 in the event of certain command errors.
```

File: cylc/flow/remote.py [hunk 2]
```diff
@@ -385,6 +388,9 @@ def remote_cylc_cmd(
     Raises:
         NoHostsError: If the platform is not contactable.
 
+    Exits:
+        1: In the event of certain command errors.
```
Comment (MetRonnie) [inline cylc/flow/remote.py:392]:
```suggestion
    Exits with code 1 in the event of certain command errors.
```

File: cylc/flow/remote.py [hunk 3]
```diff
@@ -426,6 +432,9 @@ def cylc_server_cmd(cmd, host=None, **kwargs):
...
```
Comment (MetRonnie) [inline cylc/flow/remote.py:436]:
```suggestion
    Exits with code 1 in the event of certain command errors.
```

**Pull request merged**